test/remote/gateways/remote_securion_pay_test.rb

```diff
@@ -0,0 +1,341 @@
+    assert_match %r{Requested Charge does not exist}, response.message
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+    assert_match %r{Transaction approved}, response.responses.last.message
+  end
+
+  def test_failed_verify
+    response = @gateway.verify(@declined_card, @options)
+    assert_failure response
+    assert_match %r{The card was declined for other reason.}, response.message
+    assert_match Gateway::STANDARD_ERROR_CODE[:card_declined], response.primary_response.error_code
+  end
+
+  def test_successful_store
+    assert response = @gateway.store(@credit_card, { description: "Customer Test", email: "email@example.com" })
+    assert_success response
+    assert_equal 3, response.responses.size
+    store_response = response.responses.last
+    assert_equal "customer", store_response.params["objectType"]
+    assert_equal "Customer Test", store_response.params["description"]
+    assert_equal "email@example.com", store_response.params["email"]
+    first_card = store_response.params["cards"].first
+    assert_equal store_response.params["defaultCardId"], first_card["id"]
+    assert_equal @credit_card.last_digits, first_card["last4"]
+  end
+
+  def test_successful_store_with_existing_customer
+    assert response = @gateway.store(@credit_card, { description: "Customer Test", email: "email@example.com" })
+    assert_success response
+    assert_equal 3, response.responses.size
+    store_response = response.responses.last
+
+    assert response = @gateway.store(@new_credit_card, { customer: store_response.params['id'], description: "Test Customer Update" })
+    assert_success response
+
+    assert_equal 2, response.params['cards'].size
+    first_card, second_card = response.params['cards']
+    assert_equal "card", first_card["objectType"]
+    assert_equal "card", second_card["objectType"]
+    assert_equal @new_credit_card.last_digits, second_card["last4"]
+
+    assert_equal "customer", response.params["objectType"]
+    assert_equal "Test Customer Update", response.params["description"]
+    assert_equal "email@example.com", response.params["email"]
+    assert_equal response.params["defaultCardId"], second_card['id']
+  end
+
+  def test_successful_unstore_card
+    response = @gateway.store(@credit_card, { description: "Active Merchant Unstore Customer", email: "email@example.pl" })
+    assert_success response
+    customer = response.responses.last
+    customer_id = customer.params['id']
+    card_id = customer.params['cards'].first['id']
+
+    # Unstore the card
+    assert response = @gateway.unstore(customer_id, { card_id: card_id })
+    assert_success response
+    assert_equal card_id, response.params['id']
+
+    # Unstore the customer
+    assert response = @gateway.unstore(customer_id)
+    assert_success response
+    assert_equal customer_id, response.params['id']
+  end
+
+  def test_successful_update
```

Comment:
@szajbus per my comment above, `update` as implemented does not match the correct API. Either remove or fix please.